jwst/datamodels/library.py

```diff
@@ -246,27 +246,66 @@ def get_crds_parameters(self):
-                    filename = self._temp_filenames[idx]
-                else:
-                    # otherwise, find the filename in the asn_dir
-                    member = self._members[idx]
-                    filename = Path(self._asn_dir) / member["expname"]
+            return self.read_metadata(idx)
+
+    def read_metadata(self, idx):
+        """
+        Read metadata for a model at the given index.
+
+        Parameters
+        ----------
+        idx : int
+            Index of the model in the library.
+
+        Returns
+        -------
+        dict
+            The metadata dictionary for the model.
+        """
+        # if model was already borrowed, calling code should just read the metadata
+        # from the model itself, so we raise an error here
+        if idx in self._ledger:
+            raise BorrowError("Attempt to read metadata from model that is already borrowed")
+
+        # find model in _loaded_models, temp_filenames, or asn_dir
+        if self._on_disk:
+            if idx in self._temp_filenames:
+                # if model has been modified, find its temp filename
+                filename = self._temp_filenames[idx]
             else:
-                if idx in self._loaded_models:
-                    # if this model is in memory, retrieve parameters from it directly
-                    model = self._loaded_models[idx]
-                    return model.get_crds_parameters()
-                else:
-                    # otherwise, find the filename in the asn_dir
-                    member = self._members[idx]
-                    filename = Path(self._asn_dir) / member["expname"]
+                # otherwise, find the filename in the asn_dir
+                member = self._members[idx]
+                filename = Path(self._asn_dir) / member["expname"]
+        else:
+            if idx in self._loaded_models:
+                # if this model is in memory, retrieve parameters from it directly
+                # the desired behavior is already implemented by get_crds_parameters
+                # so we can just call it here, it is not inherently CRDS-specific
+                model = self._loaded_models[idx]
+                return model.get_crds_parameters()
+            else:
+                # otherwise, find the filename in the asn_dir
+                member = self._members[idx]
+                filename = Path(self._asn_dir) / member["expname"]
+
+        meta = read_metadata(filename)
+        # update meta dict with asn-related attributes, similar to _assign_member_to_model
+        meta = self._assign_member_to_meta(meta, self._members[idx])
+        return meta
+
+    def _assign_member_to_meta(self, meta, member):
+        meta["meta.asn.exptype"] = member["exptype"]
+        for attr in ("group_id", "tweakreg_catalog"):
+            if attr in member:
+                meta["meta." + attr] = member[attr]
+        if "meta.asn" not in meta:
+            meta["meta.asn"] = {}
```

Comment:
```suggestion
        if "meta.asn" not in meta:
            meta["meta.asn"] = {}
```
Is this needed? Having `meta.asn` as a dict and a value for `meta.asn.exptype` seems inconsistent with what `get_crds_parameters` or `read_metadata` produces.

Reply:
I think I misunderstood the intent of this line in `_assign_member_to_model`.  You are right that it does not make sense here.